libs/providers/aws-ssm/src/lib/aws-ssm-provider.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  EvaluationContext,
+  Provider,
+  JsonValue,
+  ResolutionDetails,
+  StandardResolutionReasons,
+} from '@openfeature/server-sdk';
+import { AwsSsmProviderConfig } from './types';
+import { SSMService } from './ssm-service';
+import { Cache } from './cache';
+
+export class AwsSsmProvider implements Provider {
+  metadata = {
+    name: AwsSsmProvider.name,
+  };
+
+  readonly runsOn = 'server';
+  readonly service: SSMService;
+  hooks = [];
+  cache: Cache;
+
+  constructor(config: AwsSsmProviderConfig) {
+    this.service = new SSMService(config.ssmClientConfig);
+    this.cache = new Cache(config.cacheOpts);
+  }
+
+  async resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    context: EvaluationContext,
+  ): Promise<ResolutionDetails<boolean>> {
+    const cachedValue = this.cache.get(flagKey);
+    if (cachedValue) {
+      return {
+        value: cachedValue.value,
+        reason: StandardResolutionReasons.CACHED,
+      };
+    }
+    try {
+      const res = await this.service.getBooleanValue(flagKey);
+      this.cache.set(flagKey, res);
+      return res;
+    } catch (e) {
+      return {
+        value: defaultValue,
+        reason: StandardResolutionReasons.DEFAULT,
+      };
```

Comment:
```suggestion
      return {
        value: defaultValue,
        reason: StandardResolutionReasons.ERROR,
        errorCode: ErrorCode.GENERAL,
        errorMessage: err.message ?? "An unknown error occurred"
      };
```

Returning an error code here is important for telemetry purposes. The `errorMessage` can be whatever you'd like.

Reply:
You could also throw a GeneralError if you prefer. We've been shying away from throwing where possible do to the performance impact but the SDK supports either approach.

Reply:
Hi @beeme1mr , updated in https://github.com/open-feature/js-sdk-contrib/pull/1221/commits/1ccb6d9bae877dabe776a11bc962cfe092582fe7